controllers/watcher_controller.go

```diff
@@ -327,6 +329,13 @@ func (r *WatcherReconciler) Reconcile(ctx context.Context, req ctrl.Request) (re
 	// End of config generation for dbsync
 
 	ctrlResult, err := r.ensureDBSync(ctx, helper, instance, serviceLabels)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
```

Comment:
looks like we're missing a else if clause here checking for the `ctrlResult` value

Reply:
right.